Agents/NTUPVLibAgent/README.md

```diff
@@ -0,0 +1,176 @@
+
+## 3. Build and Run
+The PVLib agent can run either as a standalone docker container or within a stack. The build and setup procedures are different for different running modes.
+
+NTUPVLib is intended for deployment in a stack (option 2). For others, refer to the PVLibAgent.
+
+####  [Option 2] As a stacked docker container
+
+Running this agent in a docker stack is a more advanced option as it facilitate interactions between other agents for deployment and visualization. The stack is spun up by [Stack Manager](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager).
+
+A successful setup will result in 9 containers (optional 10):
+- Default containers
+   - Stack Manager (exits when spins up all other containers)
+   - Blazegraph
+   - Nginx
+   - Postgis
+   - Adminer
+   - Ontop
+   - Gdal
+   - Geoserver
+- PVLibAgent
+- FeatureInfoAgent (Optional)
+  Note: The FeatureInfoAgent is optional and is only required if you want to visualize the result via DTVF.
+
+##### Build the image
+First, build image with:
+```
+docker build -t ntu-pvlib-agent:1.0.0 .
+```
+The Dockerfile will automatically copy all properties files and mapping folder and set environment variables pointing to their location. Therefore, you do not need to shift the properties files and mapping folder nor add in environment variables manually.
+
+
+##### Add Config to Stack Manager
+Before running the stack manager, you need to add the config files to the stack manager. The config files are located in `TheWorldAvatar/Deploy/dynamic/stack-manager/inputs/config/`.
+- Copy `stack-manager-config/ntupvlib-agent.json` to `TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/config/services/`.
+- Create `TheWorldAvatar/Deploy/stacks/dynamic/stack-manager/inputs/config/<STACK NAME>.json` manually if it does not exist, following the below structure.
+```json
+{
+  "services": {
+    "includes": [
+      "ntupvlib-agent",
+      // Other agents you wish to spin up...
+    ],
+    "excludes": [
+      // ...
+    ]
+  }
+}
+```
+
+After this step, the stack-manager/inputs/config folder will have the following structure:
+```
+config/
+|_ services/
+   |_ ntupvlib-agent.json
+   |_ ...
+|_ <STACK NAME>.json
+```
+More information about adding custom containers to the stack can be found [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager#adding-custom-containers).
+
+##### Spin Up Stack
+Follow the [steps](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/Deploy/stacks/dynamic/stack-manager#spinning-up-a-stack) to spin up the stack.
+
+##### Run the agent
+
+Send a GET request, with the parameters including the start and end dates and times in ISO format (UTC):
+- stack=true
+- device=openmeteo
```

Comment:
will device always be openmeteo or are there other options?

Reply:
always openmeteo for the NTU case